lib/class-wp-json-comments-controller.php

```diff
@@ -0,0 +1,470 @@
+<?php
+
+/**
+ * Access comments
+ */
+class WP_JSON_Comments_Controller extends WP_JSON_Controller {
+
+	/**
+	 * Get a list of comments.
+	 *
+	 * @param  WP_JSON_Request $request Full details about the request.
+	 * @return WP_Error|WP_JSON_Response|mixed
+	 */
+	public function get_items( $request ) {
+		$prepared_args = $this->prepare_items_query( $request );
+
+		$query = new WP_Comment_Query;
+		$comments = $query->query( $prepared_args );
+
+		foreach ( $comments as &$comment ) {
```

Comment:
Looping by reference is a problem when your loop checks read permission, but doesn't drop the variable on failure :cry:

Now that I think of it, we have the same problem in the Posts controller :cry: :cry: #788